src/test/java/pl/gov/coi/pomocua/ads/users/UserResourceTest.java

```diff
@@ -0,0 +1,46 @@
+package pl.gov.coi.pomocua.ads.users;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import pl.gov.coi.pomocua.ads.TestConfiguration;
+import pl.gov.coi.pomocua.ads.UserId;
+import pl.gov.coi.pomocua.ads.authentication.TestCurrentUser;
+import pl.gov.coi.pomocua.ads.users.UsersResource.UserInfo;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Import(TestConfiguration.class)
+public class UserResourceTest {
+    @Autowired
+    protected TestRestTemplate restTemplate;
+
+    @Autowired
+    protected TestCurrentUser testCurrentUser;
+
+    @Autowired TestUsersRepository testUsersRepository;
+
+    @AfterEach
+    void clear() {
+        testUsersRepository.clear();
+    }
+
+    @Test
+    void shouldReturnCurrentUserData() {
```

Comment:
Maybe another test for missing user and returning HTTP 404?

Reply:
good point I'll set the response status to unauthorised as missing user means no account